pkg/protocolversion/factoryregistry/factoryregistry.go

```diff
@@ -52,19 +52,17 @@ func (m *Registry) CreateProtocolVersion(version string, p protocol.Protocol, ca
 }
 
 // Register registers a protocol factory for a given version
-func Register(version string, factory factory) error {
+func Register(version string, factory factory) {
 	mutex.Lock()
 	defer mutex.Unlock()
 
 	if _, ok := factories[version]; ok {
-		return fmt.Errorf("protocol version factory [%s] already registered", version)
+		panic(fmt.Errorf("protocol version factory [%s] already registered", version))
```

Comment:
panic inside the peer?

Reply:
This is on peer startup. The Fabric codebase almost always panics if the peer doesn't start properly. Besides, this situation would only occur if there was a coding error (since versions are hard-coded).